### PR TITLE
Add snowflake JDBC Connector

### PR DESCRIPTION
**Pull request by rs-alexis**
Creating a Snowflake connector

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        return super.escapeNamePattern(name, escape).toUpperCase();
+    }
+
+    @Override
+    protected ResultSet getColumns(final String catalogName, final TableName tableHandle, final DatabaseMetaData metadata)
+            throws SQLException
+    {
+        String escape = metadata.getSearchStringEscape();
+        LOGGER.warn(
+          "[SNOWFLAKE DEBUG] catalogName: {}, SchemaName: {}, TableName: {}",
```
Comment (avirtuos) [inline athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java:97]: Logging to warm but text says debug

File: athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                catalogName,
+                escapeNamePattern(tableHandle.getSchemaName(), escape),
+                escapeNamePattern(tableHandle.getTableName(), escape),
+                null);
+    }
+
+    @Override
+    public Schema getPartitionSchema(final String catalogName)
+    {
+        LOGGER.warn("[SNOWFLAKE DEBUG] getPartitionSchema ");
```
Comment (avirtuos) [inline athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java:112]: Same about log channel

File: athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    @Override
+    public GetSplitsResponse doGetSplits(BlockAllocator blockAllocator, GetSplitsRequest getSplitsRequest)
+    {
+        LOGGER.info("doGetSplits: enter - " + getSplitsRequest);
+
+        String catalogName = getSplitsRequest.getCatalogName();
+        Set<Split> splits = new HashSet<>();
+
+        Split split = Split.newBuilder(makeSpillLocation(getSplitsRequest), makeEncryptionKey()).build();
+        splits.add(split);
```
Comment (avirtuos) [inline athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java:130]: Does snowflake support any partitioning or parallelism that we can exploit to create multiple splits?

**Review by avirtuos — CHANGES_REQUESTED** [review]
Overall seems reasonable, few comments to address. Also didn't see any unit tests (maybe I missed them as I reviewed this on mobile) but both unit and integration tests will be required before we can accept this into the project as a supported connector.

**Comment by avirtuos** [comment]
Also, thanks so much for the contribution!

**Comment by rs-alexis** [comment]
> Overall seems reasonable, few comments to address. Also didn't see any unit tests (maybe I missed them as I reviewed this on mobile) but both unit and integration tests will be required before we can accept this into the project as a supported connector.

@avirtuos Thanks for your feedback, This is still a draft no the final PR yet
I'll add Unit Tests

File: athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java (hunk 1 above)
Comment (rs-alexis) [reply athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java:97]: Yeah the PR is not ready yet, shouldn't have pushed my debug lol

File: athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java (hunk 3 above)
Comment (rs-alexis) [reply athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java:130]: That's the big question, Snowflake supports micro-partitioning but not sure yet how that would fit in the structure of the code yet (https://docs.snowflake.com/en/user-guide/tables-clustering-keys.html)

I'm open to feedback on that

**Comment by avirtuos** [comment]
is this ready for final review? or are you still working on it?

**Comment by rs-sathya** [comment]
@avirtuos Yes, this should be ready for final review btw

**Comment by henrymai** [comment]
Thank you for this contribution but we now have a snowflake connector: https://github.com/awslabs/aws-athena-query-federation/tree/master/athena-snowflake

If there is a feature in this pull request that is missing from the connector above, please open a new pull request to update the referenced connector. Thanks.

**Pull request closed without merging**